sockeye/training.py

```diff
@@ -146,657 +130,848 @@ def sym_gen(seq_lens):
+    def _save_initial_optimizer_states(self, lr_decay_opt_states_reset: str):
+        if lr_decay_opt_states_reset == C.LR_DECAY_OPT_STATES_RESET_INITIAL:
+            self.model.save_optimizer_states(os.path.join(self.model.output_dir, C.OPT_STATES_INITIAL))
+
+    def _check_args(self,
+                    metrics: List[str],
+                    early_stopping_metric: str,
+                    lr_decay_opt_states_reset: str,
+                    lr_decay_param_reset: bool,
+                    cp_decoder: Optional[checkpoint_decoder.CheckpointDecoder] = None):
+        """
+        Helper function that checks various configuration compatibilities.
+        """
+        utils.check_condition(early_stopping_metric in metrics, "Early stopping metric must be tracked.")
+        utils.check_condition(len(metrics) > 0, "At least one metric must be provided.")
+        for metric in metrics:
+            utils.check_condition(metric in C.METRICS, "Unknown metric to track during training: %s" % metric)
+
+        if 'dist' in self.optimizer_config.kvstore:
+            # In distributed training the optimizer will run remotely. For eve we however need to pass information about
+            # the loss, which is not possible anymore by means of accessing self.module._curr_module._optimizer.
+            utils.check_condition(self.optimizer_config.name != C.OPTIMIZER_EVE,
+                                  "Eve optimizer not supported with distributed training.")
+            utils.check_condition(
+                not issubclass(type(self.optimizer_config.lr_scheduler), lr_scheduler.AdaptiveLearningRateScheduler),
+                "Adaptive learning rate schedulers not supported with a dist kvstore. "
+                "Try a fixed schedule such as %s." % C.LR_SCHEDULER_FIXED_RATE_INV_SQRT_T)
+            utils.check_condition(not lr_decay_param_reset, "Parameter reset when the learning rate decays not "
+                                                            "supported with distributed training.")
+            utils.check_condition(lr_decay_opt_states_reset == C.LR_DECAY_OPT_STATES_RESET_OFF,
+                                  "Optimizer state reset when the learning rate decays "
+                                  "not supported with distributed training.")
+
+        utils.check_condition(self.optimizer_config.gradient_clipping_type in C.GRADIENT_CLIPPING_TYPES,
+                              "Unknown gradient clipping type %s" % self.optimizer_config.gradient_clipping_type)
+
+        utils.check_condition(early_stopping_metric in C.METRICS,
+                              "Unsupported early-stopping metric: %s" % early_stopping_metric)
+        if early_stopping_metric == C.BLEU:
+            utils.check_condition(cp_decoder is not None, "%s requires CheckpointDecoder" % C.BLEU)
+
+    def _check_dist_kvstore_requirements(self, lr_decay_opt_states_reset:bool, lr_decay_param_reset, optimizer):
```

Comment:
maybe github is hiding something. Is this still called?

Reply:
no, and its not needed anymore since this logic is in `_check_args()`. Will remove `_check_dist_kvstore_requirements`